cmd/clients.go

```diff
@@ -34,8 +36,19 @@ type Clients struct {
 // PrepareClients instantiates all the necessary clients for a validator node
 func PrepareClients(config config.Clients) *Clients {
 	EVMClients := make(map[uint64]client.EVM)
-	for chainId, ec := range config.Evm {
-		EVMClients[chainId] = evm.NewClient(ec)
+	for configChainId, ec := range config.Evm {
+		EVMClients[configChainId] = evm.NewClient(ec, configChainId)
+		clientChainId, e := EVMClients[configChainId].ChainID(context.Background())
+		if e != nil {
+			log.Fatalf("[%d] - Failed to retrieve chain ID on client prepare.", configChainId)
+			return nil
```

Comment:
l`og.Fatalf` exits the program There is no need to return anything.

Reply:
done